tests/golden/test_schedules/test_fail_stage_mem.txt

```diff
@@ -0,0 +1,15 @@
+def sqmat(n: size, A: R[n, n] @ DRAM, B: R[n, n] @ DRAM):
+    assert n % 4 == 0
+    for i in seq(0, n / 4):
+        for j in seq(0, n / 4):
+            for k in seq(0, n / 4):
+                Btile: R[4, 4] @ DRAM
+                for i0 in seq(0, 4):
+                    for i1 in seq(0, 4):
+                        Btile[i0, i1] = B[i0 + 4 * i, i1 + 4 * k]
+                for ii in seq(0, 4):
+                    for jj in seq(0, 4):
+                        for kk in seq(0, 4):
+                            A[ii + 4 * i,
+                              jj + 4 * j] += Btile[ii, kk] * B[kk + 4 * k,
```

Comment:
I think there's still a bug. This replace should fail because there are cases where the second read of `B` could overlap with the first read, e.g. if `i = j = k` and `ii = jj = kk`. In this particular test case, it's not fatal because `B` is read only, but if you add a `B[kk + 4 * k, jj + 4 * j] = 1.0` statement inside the loop, then `stage_mem` will break correctness.